src/github.com/getlantern/autoupdate-server/server/asset.go

```diff
@@ -0,0 +1,60 @@
+package server
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+)
+
+const (
+	assetsDirectory = "assets/"
+)
+
+func init() {
+	err := os.MkdirAll(assetsDirectory, os.ModeDir|0700)
+	if err != nil {
+		log.Fatalf("Could not create directory for storing assets: %q", err)
+	}
+}
+
+// downloadAsset grabs the contents of the body of the given URL and stores
+// then into $ASSETS_DIRECTORY/$BASENAME.SHA256_SUM($URL)
+func downloadAsset(uri string) (localfile string, err error) {
+	basename := path.Base(uri)
+
+	if len(basename) > 60 {
+		basename = basename[:60]
+	}
```

Comment:
That line looks suspicious indeed! I added a comment to clarify what's the intention of having it.